File: PHP/Objects/HerokuWeb.php
<?php
use Facebook\WebDriver\Remote\RemoteWebDriver;
use Facebook\WebDriver\WebDriverBy;
use Facebook\WebDriver\WebDriverExpectedCondition;

class HerokuWeb {
  private $driver_, $loginMsg, $url, $username, $password, $usernameId, $passwordId ,$loginMessageId, $formXpath;

  function __construct($driver) {
    $this->driver_ = $driver;
    $this->loginMsg = "";
    $this->url = "https://the-internet.herokuapp.com/login";
    $this->username = "tomsmith";
    $this->password = "SuperSecretPassword!";
    $this->usernameId = "username";
    $this->passwordId = "password";
    $this->loginMessageId = "flash";
    $this->formXpath = "//form[@name='login']";
  }

  private function login() {
    $this->driver_->get($this->url);
    $elementUsername = $this->driver_->findElement(WebDriverBy::id($this->usernameId));
    $this->waitUntilVisible($elementUsername);
    $elementUsername->sendKeys($this->username);

    $elementPassword = $this->driver_->findElement(WebDriverBy::id($this->passwordId));
    $this->waitUntilVisible($elementPassword);
    $elementPassword->sendKeys($this->password);

    $elementForm = $this->driver_->findElement(WebDriverBy::xpath($this->formXpath));
    $this->waitUntilVisible($elementForm);
    $elementForm->submit();
  }

  public function getLoginMessage() {
    $this->login();
    $elementLoginMsg = $this->driver_->findElement(WebDriverBy::id($this->loginMessageId));
    $this->waitUntilVisible($elementLoginMsg);
    return $elementLoginMsg->getText();
  }

  private function waitUntilVisible($element) {
    // Default wait() (= 30 sec)
    // Wait for at most 10s and retry every 500ms if it the title is not correct.
    $this->driver_->wait(10, 500)->until(WebDriverExpectedCondition::visibilityOf($element));
  }
}
?>

File: Ruby/Objects/HerokuWeb.rb
require 'selenium-webdriver'
require_relative '../Utils/Driver'
require_relative '../Utils/Helpers'

class HerokuWeb
  def initialize driver
    @driver = driver
    @loginMsg = ''
    @url = 'https://the-internet.herokuapp.com/login'
    @username = 'tomsmith'
    @password = 'SuperSecretPassword!'
    @usernameId = 'username'
    @passwordId = 'password'
    @loginMessageId = 'flash'
    @btnLoginXpath = '//button[contains(.,"Login")]'
    @waitingTime = 20
    @pageTitle = 'the internet'
  end

  def login
    @driver.navigate.to @url
    wait_for(@waitingTime) {@driver.title.downcase.start_with? @pageTitle}
    @driver.find_element(:id, @usernameId).send_keys @username
    @driver.find_element(:id, @passwordId).send_keys @password
    @driver.find_element(:xpath, @btnLoginXpath).click
  end

  def getLoginMessage
    return @driver.find_element(:id, @loginMessageId).text
  end

end
